apiserver/facades/client/controller/controller.go

```diff
@@ -544,40 +488,25 @@ func (c *ControllerAPI) HostedModelConfigs(ctx context.Context) (params.HostedMo
 		return result, errors.Trace(err)
 	}
 
-	modelUUIDs, err := c.state.AllModelUUIDs()
+	hostedModels, err := c.modelService.HostedModels(ctx)
 	if err != nil {
-		return result, errors.Trace(err)
+		return result, fmt.Errorf("getting hosted models: %w", err)
 	}
 
-	for _, modelUUID := range modelUUIDs {
-		if modelUUID == c.state.ControllerModelUUID() {
-			continue
-		}
-		st, err := c.statePool.Get(modelUUID)
-		if err != nil {
-			// This model could have been removed.
-			if errors.Is(err, errors.NotFound) {
-				continue
-			}
-			return result, errors.Trace(err)
-		}
-		defer st.Release()
-		model, err := st.Model()
-		if err != nil {
-			return result, errors.Trace(err)
-		}
-
+	for _, hostedModel := range hostedModels {
 		config := params.HostedModelConfig{
-			Name:     model.Name(),
-			OwnerTag: model.Owner().String(),
+			Name:     hostedModel.Model.Name,
+			OwnerTag: names.NewUserTag(hostedModel.Model.OwnerName).String(),
 		}
-		modelConf, err := model.Config()
+
+		modelConf, err := c.modelConfigServiceGetter(hostedModel.Model.UUID).ModelConfig(ctx)
```

Comment:
Why is the modelConfigService being used as well here? Since you're adding a HostedModels method to the models service can this not just get exactly the information from the db that you need for this apiserver method?

Reply:
Ah, after reading Joe and Simons comments I understand why this is here/what it is doing.